src/staking-v3/hooks/useDappStaking.ts

```diff
@@ -445,15 +445,6 @@ export function useDappStaking() {
         return [false, t('stakingV3.dappStaking.WrongNetworkZkEvm'), ''];
       } else if (stake.amount <= 0) {
         return [false, t('stakingV3.dappStaking.ZeroAmount'), ''];
-      } else if (
-        constants.value &&
-        (ledger.value?.contractStakeCount ?? 0) >= constants.value.maxNumberOfStakedContracts
-      ) {
-        return [
-          false,
-          t('stakingV3.dappStaking.TooManyStakedContracts'),
-          docsUrl.dappStakingStaked16Dapps,
-        ];
```

Comment:
Just informative - the check isn't completely wrong, just incomplete.
You should also check that the _dapp_ on which the user is trying to stake doesn't have an existing DB entry for the user.

If it does, the call will work, because existing entry is being modified, which won't increase number of staked contracts.
If a new contract is being staked on, one for which the staker doesn't have an entry in the DB, then the call will fail.